Automattic/wp-calypso
# Change Plugins route logic and move Plugins components from PluginsBrowser

#### Proposed Changes

* Extracts `PluginsCategoryResults` and  page from `PluginsBrowser`.
* Make `PluginsSearchResults`, `PluginsDiscoveryPage`, and `PluginsCategoryResults` components to work independently.
* Routes controller pass `PluginsSearchResults`, `PluginsDiscoveryPage`, and `PluginsCategoryResults` components to `PluginsBrowser`.
* Clean code from `PluginsBrowser`.

#### Testing Instructions

1. Go to `/plugins`.
2. Check `Discovery` to show correct elements (`Top premium plugins`, `Editor's pick`, and `Top free plugins`).
3. Filter for a Category and check for correct results.
4. Search or click on any Popular Search and check for correct results.

#### Pre-merge Checklist

Fixes #64547
Fixes #64548
